### ISSUE TITLE
Use z-index to traverse the semantics object.

### ISSUE THREAD

**Issue by hangyujin** [flutter/flutter#118656]
## Use case
A follow-up on https://github.com/flutter/engine/pull/38794 

Currently for overlapping semantics object in ios, the focus goes to the smallest item, the ideal way should be use z-index from framework to focus.
